### PR TITLE
Give some intrinsics fallback bodies

### PR DESCRIPTION
**Pull request by oli-obk**
cc #93145

### CODE REVIEW

File: library/core/src/intrinsics.rs [hunk 1]
```diff
@@ -1065,33 +1081,38 @@ extern "rust-intrinsic" {
...
-    pub fn assert_inhabited<T>();
+/// A guard for unsafe functions that cannot ever be executed if `T` is uninhabited:
+/// This will statically either panic, or do nothing.
+///
+/// This intrinsic does not have a stable counterpart.
+#[rustc_const_stable(feature = "const_assert_type", since = "1.59.0")]
+#[rustc_nounwind]
+#[unstable(feature = "core_intrinsics", issue = "none")]
+#[cfg_attr(not(bootstrap), rustc_intrinsic)]
+pub const fn assert_inhabited<T>() {}
```
Comment (WaffleLapkin) [inline library/core/src/intrinsics.rs:1094]: I don't think that's right? I presume unsafe function can depend on this (and other `intrinsic::assert_*`) panicking and so doing nothing always is unsound?

**Comment by WaffleLapkin** [comment]
r? WaffleLapkin

File: library/core/src/intrinsics.rs (hunk 1 above)
Comment (oli-obk) [reply library/core/src/intrinsics.rs]: oh, right 🤦 thanks

**Review by WaffleLapkin — APPROVED** [review]

**Pull request merged**